core/client/scenes/scene-ui.js

```diff
@@ -106,6 +106,13 @@ UIScene = new Phaser.Class({
     });
   },
 
+  updateUserIcon(userId, icon) {
+    const textInstance = this.characterNamesObjects[userId];
+    if (textInstance) {
```

Comment:
Why is that check necessary?

Reply:
Theoretically it shouldn't be false, I've let it as a safeguard, if for some reasons updateUserIcon is called before a name was set.